runtime/util/hshelp.c

```diff
@@ -3229,6 +3230,8 @@ classIsModifiable(J9JavaVM * vm, J9Class * clazz)
 		rc = JNI_FALSE;
 	} else if ((J2SE_VERSION(vm) >= J2SE_19) && J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsAnonymous)) {
 		rc = JNI_FALSE;
+	} else if (strcmp(J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass)), "com/ibm/jit/JITHelpers") == 0) {
```

Comment:
`J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass))` isn't guaranteed to be null terminated.

We need to check the length matches the length of `com/ibm/jit/JITHelpers` before doign the strcmp

Reply:
my fix checks that the class name length is either 22 (com/ibm/jit/JITHelper without terminator) or 23 (com/ibm/jit/JITHelper with terminator) before comparing characters